functional/FunctionalTest.java

```diff
@@ -749,16 +749,31 @@ public static void testPutObject(String testTags, PutObjectArgs args, String err
       throws Exception {
     String methodName = "putObject()";
     long startTime = System.currentTimeMillis();
+    ObjectWriteResponse objectInfo = null;
     try {
       try {
-        client.putObject(args);
+        objectInfo = client.putObject(args);
       } catch (ErrorResponseException e) {
         if (errorCode == null || !e.errorResponse().code().equals(errorCode)) {
           throw e;
         }
       }
+      if (testTags.equals("[With Retention]")) {
+        client.setObjectRetention(
+            SetObjectRetentionArgs.builder()
+                .bucket(args.bucket())
+                .object(args.object())
+                .config(new Retention())
+                .bypassGovernanceMode(true)
+                .build());
+      }
       client.removeObject(
-          RemoveObjectArgs.builder().bucket(args.bucket()).object(args.object()).build());
+          RemoveObjectArgs.builder()
+              .bucket(args.bucket())
+              .object(args.object())
+              .versionId(objectInfo != null ? objectInfo.versionId() : null)
```

Comment:
Just use `.versionId(objectInfo.versionId())`

Reply:
It wont work as below test doesn't populate objectInfo
```
 testPutObject(
        "[storage-class=INVALID negative case]",
        PutObjectArgs.builder().bucket(bucketName).object(getRandomName()).stream(
                new ContentInputStream(1 * KB), 1 * KB, -1)
            .headers(headers)
            .build(),
        "InvalidStorageClass");

```